time ts
uint8[1206] buf
uint32 size